api/http/common.go

```diff
@@ -96,6 +100,19 @@ func ValidateUUID(extID string) (err error) {
 	return nil
 }
 
+// ValidateName validates name format.
+func ValidateName(id string) error {
+	if !nameRegExp.MatchString(id) {
+		return apiutil.ErrInvalidNameFormat
+	}
+	// Names containing double underscores or double dashes are invalid due to similarity concerns.
+	if strings.Contains(id, "__") || strings.Contains(id, "--") {
+		return apiutil.ErrInvalidNameFormat
```

Comment:
We need to wrap this to make it more precise so the user knows what's wrong with the name format.